lib/facter/self_service.rb

```diff
@@ -130,4 +130,9 @@
     # Is Hiera 5 in use?
     { S0033: hiera_config_file.dig('version') == 5 }
   end
+
+  chunk(:S0035) do
+    # return false if any Warnings appear in the 'puppet module list...'
```

Comment:
This can be added to restrict S0035 to puppetservers

```
next unless PuppetSelfService.replica? || PuppetSelfService.compiler? || PuppetSelfService.legacy_compiler? || PuppetSelfService.primary?
```